Trying to access settings while onboarding triggers crash

Tapping `Settings` on the first onboarding screen (the one with the `Let's go` button) triggers the following crash:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{dev.msfjarvis.aps.debug/dev.msfjarvis.aps.ui.settings.SettingsActivity}: java.lang.IllegalStateException: Setting the checked value before the preference was attached isn't supported. Consider using `defaultValue` instead.
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
	at android.app.ActivityThread.-wrap11(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6494)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: java.lang.IllegalStateException: Setting the checked value before the preference was attached isn't supported. Consider using `defaultValue` instead.
	at de.Maxr1998.modernpreferences.preferences.TwoStatePreference.setChecked(TwoStatePreference.kt:30)
	at dev.msfjarvis.aps.ui.settings.GeneralSettings.provideSettings(GeneralSettings.kt:69)
	at dev.msfjarvis.aps.ui.settings.SettingsActivity.onCreate(SettingsActivity.kt:38)
	at android.app.Activity.performCreate(Activity.java:7009)
	at android.app.Activity.performCreate(Activity.java:7000)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
	... 9 more
```

_Originally posted in https://github.com/android-password-store/Android-Password-Store/pull/1370#issuecomment-813866168_

--- a/app/src/main/java/dev/msfjarvis/aps/ui/settings/GeneralSettings.kt
+++ b/app/src/main/java/dev/msfjarvis/aps/ui/settings/GeneralSettings.kt
@@ -58,49 +58,43 @@ class GeneralSettings(private val activity: FragmentActivity) : SettingsProvider
         defaultValue = false
       }
 
+      val canAuthenticate = BiometricAuthenticator.canAuthenticate(activity)
       checkBox(PreferenceKeys.BIOMETRIC_AUTH) {
         titleRes = R.string.pref_biometric_auth_title
         defaultValue = false
-      }
-        .apply {
-          val canAuthenticate = BiometricAuthenticator.canAuthenticate(activity)
-          if (!canAuthenticate) {
-            enabled = false
-            checked = false
-            summaryRes = R.string.pref_biometric_auth_summary_error
-          } else {
-            summaryRes = R.string.pref_biometric_auth_summary
-            onClick {
-              enabled = false
-              val isChecked = checked
-              activity.sharedPrefs.edit {
-                BiometricAuthenticator.authenticate(activity) { result ->
-                  when (result) {
-                    is BiometricAuthenticator.Result.Success -> {
-                      // Apply the changes
-                      putBoolean(PreferenceKeys.BIOMETRIC_AUTH, checked)
-                      enabled = true
-                    }
-                    else -> {
-                      // If any error occurs, revert back to the previous
-                      // state. This
-                      // catch-all clause includes the cancellation case.
-                      putBoolean(PreferenceKeys.BIOMETRIC_AUTH, !checked)
-                      checked = !isChecked
-                      enabled = true
-                    }
-                  }
+        enabled = canAuthenticate
+        summaryRes =
+          if (canAuthenticate) R.string.pref_biometric_auth_summary else R.string.pref_biometric_auth_summary_error
+        onClick {
+          enabled = false
+          val isChecked = checked
+          activity.sharedPrefs.edit {
+            BiometricAuthenticator.authenticate(activity) { result ->
+              when (result) {
+                is BiometricAuthenticator.Result.Success -> {
+                  // Apply the changes
+                  putBoolean(PreferenceKeys.BIOMETRIC_AUTH, checked)
+                  enabled = true
+                }
+                else -> {
+                  // If any error occurs, revert back to the previous
+                  // state. This
+                  // catch-all clause includes the cancellation case.
+                  putBoolean(PreferenceKeys.BIOMETRIC_AUTH, !checked)
+                  checked = !isChecked
+                  enabled = true
                 }
               }
-              if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
-                activity.getSystemService<ShortcutManager>()?.apply {
-                  removeDynamicShortcuts(dynamicShortcuts.map { it.id }.toMutableList())
-                }
-              }
-              false
             }
           }
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+            activity.getSystemService<ShortcutManager>()?.apply {
+              removeDynamicShortcuts(dynamicShortcuts.map { it.id }.toMutableList())
+            }
+          }
+          false
         }
+      }
     }
   }
 }

